container_opts.go

```diff
@@ -154,14 +154,18 @@ func setSnapshotterIfEmpty(c *containers.Container) {
 // to encode and decode the object.
 func WithContainerExtension(name string, extension interface{}) NewContainerOpts {
 	return func(ctx context.Context, client *Client, c *containers.Container) error {
+		if name == "" {
+			return errors.Wrapf(errdefs.ErrInvalidArgument, "extension key must not be zero-length")
+		}
+
 		any, err := typeurl.MarshalAny(extension)
 		if err != nil {
-			return err
+			if errors.Cause(err) == typeurl.ErrNotFound {
+				return errors.Wrap(err, "extension is not registered with the typeurl package and cannot be marshalled, make sure to use `typeurl.Register` for the passed in extension type")
```

Comment:
nit: maybe a tad to verbose? I would reduce it to `extension is not registered with the typeurl package`

Reply:
Pushed this... basically, `errors.Wrapf(err, "extension %q is not registered with the typeurl package, see `typeurl.Register`", name)`

Reply:
Wondering if we should provide a helper since `typeurl` is not really used anywhere else in the client from what I've seen.

Reply:
Perfect, thanks.